wordpress-mobile/WordPress-Android
# Refactors `Notes` to Kotlin

## Description
This PR:
1. Converts the `Notes` model to Kotlin
2. Removes unused code
3. Replaces getters with lazy initialisation to limit json queries
4. Removes synchronisation locks
5. Cleans up the code and adds Kotlin idioms when possible

-----

## To Test:
1. Open the notifications tab
2. Scroll/Open notifications
3. Like/Share notifications
4. Change between tabs
5. *Verify* that there are no changes in the behaviour of the app

-----

## Regression Notes

1. Potential unintended areas of impact

    - Notifications

7. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

8. What automated tests I added (or what prevented me from doing so)

    - None at this stage

-----

## PR Submission Checklist:


-----

## Testing Checklist:
